tests/validators/allow_empty_string_test.py

```diff
@@ -0,0 +1,110 @@
+"""
+validataclass
+Copyright (c) 2021, binary butterfly GmbH and contributors
+Use of this source code is governed by an MIT-style license that can be found in the LICENSE file.
+"""
+
+from decimal import Decimal
+
+import pytest
+from validataclass.exceptions import ValidationError
+from validataclass.validators import AllowEmptyString, DecimalValidator, IntegerValidator
+from tests.test_utils import UnitTestContextValidator
+
+
+class AllowEmptyStringTest:
+    """
+    Unit tests for the AllowEmptyString wrapper validator.
+    """
+
+    @staticmethod
+    @pytest.mark.parametrize(
+        'input_data, expected_result',
+        [
+            ('', ''),
+            ('12.34', Decimal('12.34')),
+        ]
+    )
+    def test_allow_empty_string_valid(input_data, expected_result):
+        """ Test AllowEmptyString with different valid input (empty string ('') and not empty string). """
+        validator = AllowEmptyString(DecimalValidator())
+        result = validator.validate(input_data)
+
+        assert type(result) == type(expected_result)
+        assert result == expected_result
+
+    @staticmethod
+    @pytest.mark.parametrize(
+        'input_data, expected_result',
+        [
+            ('', Decimal('3.1415')),
+            ('12.34', Decimal('12.34')),
+        ]
+    )
+    def test_allow_empty_string_with_default_valid(input_data, expected_result):
+        """ Test AllowEmptyString with a custom default value with different valid input (empty string ('') and not empty string). """
+        validator = AllowEmptyString(DecimalValidator(), default=Decimal('3.1415'))
+        result = validator.validate(input_data)
+
+        assert type(result) == type(expected_result)
+        assert result == expected_result
+
+    @staticmethod
+    def test_allow_empty_string_with_context_arguments():
+        """ Test that AllowEmptyString passes context arguments down to the wrapped validator. """
+        validator = AllowEmptyString(UnitTestContextValidator())
+        assert validator.validate('') is ''
```

Comment:
This is the line flake8 complains about (see test suite, or run `make tox` or `make flake8` locally):

> tests/validators/allow_empty_string_test.py:56:16: F632 use ==/!= to compare constant literals (str, bytes, int, float, tuple)

Reply:
Changed